types/sparqljs/index.d.ts

```diff
@@ -261,19 +286,19 @@ export interface Triple {
 
 export interface PropertyPath {
     type: "path";
-    pathType: "|" | "/" | "^" | "+" | "*" | "!";
+    pathType: "|" | "/" | "^" | "+" | "*" | "!" | "?";
     items: Array<IriTerm | PropertyPath>;
 }
 
 export type Expression =
     | OperationExpression
     | FunctionCallExpression
     | AggregateExpression
-    | BgpPattern
-    | GraphPattern
-    | GroupPattern
-    | Tuple
-    | Term;
+    | Pattern // all patterns can be inside an exists.
```

Comment:
Maybe it would be better to explicitly add `ExistsExpression` as a separate union branch here instead of allowing any pattern to be an expression (which is usually not the case)?

Reply:
I agree that would be even better, but given the state of the current types (which are plainly wrong), this is at least correct.

Reply:
That's true, however may I suggest I less breaking change from type perspective?
```ts
export interface OperationExpression extends BaseExpression {
    type: "operation";
    operator: string;
    args: Array<Expression | Pattern>;
}
```
This way it could be later in the future refined by converting into a disjoint union type on `operator` property if it would be desirable.

Reply:
Nice solution!